persist/sqlite/migrations.go

```diff
@@ -12,6 +12,73 @@ import (
 	"go.uber.org/zap"
 )
 
+// migrateVersion45 cleans up unresolved v1 contracts since they will never complete
+// and v1 support has been removed.
+func migrateVersion45(tx *txn, log *zap.Logger) error {
+	rows, err := tx.Query(`SELECT id FROM contracts WHERE contract_status IN ($1, $2)`, contracts.ContractStatusActive, contracts.ContractStatusPending)
+	if err != nil {
+		return fmt.Errorf("failed to query active and pending contracts: %w", err)
+	}
+	defer rows.Close()
+
+	var contractIDs []int64
+	for rows.Next() {
+		var id int64
+		if err := rows.Scan(&id); err != nil {
+			return fmt.Errorf("failed to scan contract ID: %w", err)
+		}
+		contractIDs = append(contractIDs, id)
+	}
+	if err := rows.Err(); err != nil {
+		return fmt.Errorf("failed to iterate contract IDs: %w", err)
+	} else if err := rows.Close(); err != nil {
+		return fmt.Errorf("failed to close rows: %w", err)
+	} else if len(contractIDs) == 0 {
+		return nil
+	}
+
+	deleteSectorsStmt, err := tx.Prepare(`DELETE FROM contract_sector_roots WHERE contract_id = ?`)
+	if err != nil {
+		return fmt.Errorf("failed to prepare delete statement: %w", err)
+	}
+	defer deleteSectorsStmt.Close()
+
+	deleteContractAccountFundingStmt, err := tx.Prepare(`DELETE FROM contract_account_funding WHERE contract_id = ?`)
+	if err != nil {
+		return fmt.Errorf("failed to prepare delete statement: %w", err)
+	}
+	defer deleteContractAccountFundingStmt.Close()
+
+	deleteContractsStmt, err := tx.Prepare(`DELETE FROM contracts WHERE id = ?`)
+	if err != nil {
+		return fmt.Errorf("failed to prepare delete statement: %w", err)
+	}
+	defer deleteContractsStmt.Close()
+
+	for _, id := range contractIDs {
+		if _, err := deleteSectorsStmt.Exec(id); err != nil {
+			return fmt.Errorf("failed to delete contract sector roots: %w", err)
+		} else if _, err := deleteContractAccountFundingStmt.Exec(id); err != nil {
+			return fmt.Errorf("failed to delete contract account funding: %w", err)
+		} else if _, err := deleteContractsStmt.Exec(id); err != nil {
+			return fmt.Errorf("failed to delete contracts: %w", err)
+		}
+	}
+
+	if err := recalcContractAccountFunding(tx, log); err != nil {
```

Comment:
Do you need to call this? I guess better safe than sorry but at this point the contract and its account funding was deleted and it shouldn't have changed for other contracts?